src/components/PlanGate.tsx

```diff
@@ -66,7 +66,7 @@ export function PlanGate ({ children }: { children: ReactNode }): ReactNode {
 
 export function MaybePlanGate ({ children }: { children: ReactNode }): ReactNode {
   const params = useSearchParams()
-  if ((process.env.NEXT_PUBLIC_DISABLE_PLAN_GATE == 'true') || (params.get('checkout') === 'true')) {
+  if ((process.env.NEXT_PUBLIC_DISABLE_PLAN_GATE == 'true') || params.get('checkout.session')) {
```

Comment:
Should we check if the checkout was completed?
e.g: https://docs.stripe.com/api/events/types#event_types-checkout.session.completed

Reply:
ah so that's a slightly separate thing - the name matches the event names, but the param is actually just set manually in the Stripe pricing table config: 

<img width="559" alt="Screenshot 2024-12-12 at 10 39 15 AM" src="https://github.com/user-attachments/assets/28b12033-4ac4-47bf-89b9-6903cae67abd" />

I had been planning on adding `checkout` as a custom query param but realized we already have `checkout.session` - we only redirect here if checkout completed so it's essentially what you're suggesting already